### PR TITLE
Clarify why `hosts` in `daemon.json` does not work

### PR DESCRIPTION
**Pull request by albers**
Ref https://github.com/docker/docker/issues/30996#issuecomment-281299320

### CODE REVIEW

**Review by thaJeztah — APPROVED** [review]
LGTM, thanks!

File: docs/reference/commandline/dockerd.md [hunk 1]
```diff
@@ -1208,6 +1208,13 @@ This is a full example of the allowed configuration options on Linux:
 }
 ```
 
+> **Note:** You cannot set options in `daemon.json` that have already been set on
+> daemon startup as a flag.
+> On systems that use `systemd` to start the Docker daemon, `-H` is already set, so
+> you cannot use the `host` key in `daemon.json` to add listening addresses.
```
Comment (albers) [inline docs/reference/commandline/dockerd.md]: Should be `hosts`, will fix.

Comment (thaJeztah) [inline docs/reference/commandline/dockerd.md]: oh, you're right, thanks for the catch!

**Review by cpuguy83 — APPROVED** [review]
LGTM

**Pull request merged**